staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -623,10 +653,63 @@ func preprocessDeletionListForMerging(key string, original map[string]interface{
 	return false, false, "", nil
 }
 
+// processReplaceKeys processes replaceKeys list.
+// When MergeParallelList is true, we validate the patch and
+// then clear the fields that are not in the list.
+func processReplaceKeys(original, patch map[string]interface{}, options MergeOptions) error {
```

Comment:
// processRetainKeys looks for a retainkeys directive and applies to original
// - if no directive exists do nothing
// - if directive is found, clear keys in original missing from the directive list
// - validate that all keys present in the patch are present in the retainkeys directive